doc/README.md

```diff
@@ -188,24 +188,24 @@ In that case, please open an issue about the particular documentation convention
   }
   ```
 
-- Use [definition lists](#definition-lists) to document function arguments, and the attributes of such arguments. For example:
+- Use [definition lists](#definition-lists) to document function arguments, and the attributes of such arguments as well as their [types](https://nixos.org/manual/nix/stable/language/values).
+  For example:
 
   ```markdown
   # pkgs.coolFunction
 
   Description of what `coolFunction` does.
   `coolFunction` expects a single argument which should be an attribute set, with the following possible attributes:
 
-  `name`
+  `name` (String)
 
   : The name of the resulting image.
 
-  `tag` _optional_
+  `tag` (String, _optional_)
```

Comment:
```suggestion
  `tag` (String; _optional_)
```

When there's more than one type allowed, sometimes I have to use a comma to separate them, so I started using a semicolon to separate the types from additional properties. Example: `Path, List of Paths, or Null; _optional_`.

Also open to suggestions on a different approach.

Reply:
"Around here" we lean towards the Haskell type notation e.g. `(String | Integer)`, but we probably shouldn't use it without explanation. I've seen multiple exchanges where people discussed which notation to use and how to introduce it to readers and maintainers. From experience, I suggest not thinking too hard about it in one place, as that will just prevent any user-facing progress at all, and instead evolve it incrementally.